docs/book/src/tasks/automated-machine-management/machine_deletions.md

```diff
@@ -0,0 +1,127 @@
+# Machine deletion process
+
+Machine deletions occur in various cases, for example:
+* Control plane (e.g. KCP) or MachineDeployment rollouts
+* Machine remediations
+* Scale downs of MachineDeployments
+
+This page describes how Cluster API deletes Machines.
+
+Machine deletion can be broken down into the following phases:
+1. Machine deletion is triggered (i.e. the `metadata.deletionTimestamp` is set)
+2. Machine controller waits until all pre-drain hooks succeeded, if any are registered
+    * Pre-drain hooks can be registered by adding annotations with the `pre-drain.delete.hook.machine.cluster.x-k8s.io` prefix to the Machine object
+3. Machine controller checks if the Machine should be drained, drain is skipped if:
+    * The Machine has the `machine.cluster.x-k8s.io/exclude-node-draining` annotation
+    * The `Machine.spec.nodeDrainTimeout` field is set and already expired (unset or `0` means no timeout)
+4. If the Machine should be drained, the Machine controller evicts all relevant Pods from the Node (see details in [Node drain](#node-drain))
+5. Machine controller checks if we should wait until all volumes are detached, this is skipped if:
+    * The Machine has the `machine.cluster.x-k8s.io/exclude-wait-for-node-volume-detach` annotation
+    * The `Machine.spec.nodeVolumeDetachTimeout` field is set and already expired (unset or `0` means no timeout)
+6. If we should wait for volume detach, the Machine controller waits until `Node.status.volumesAttached` is empty
+    * Typically the volumes are getting detached by CSI after the corresponding Pods have been evicted during drain
+7. Machine controller waits until all pre-terminate hooks succeeded, if any are registered
+    * Pre-terminate hooks can be registered by adding annotations with the `pre-terminate.delete.hook.machine.cluster.x-k8s.io` prefix to the Machine object
+8. Machine controller deletes the `InfrastructureMachine` object (e.g. `DockerMachine`) of the Machine and waits until it is gone
+9. Machine controller deletes the `BootstrapConfig` object (e.g. `KubeadmConfig`) of the machine and waits until it is gone
+10. Machine controller deletes the Node object in the workload cluster
+    * Node deletion will be retried until either the Node object is gone or `Machine.spec.nodeDeletionTimeout` is expired (`0` means no timeout, but the field defaults to 10s)
+
+Note: There are cases where Node drain, wait for volume detach and Node deletion is skipped. For these please take a look at the 
+implementation of the [`isDeleteNodeAllowed` function](https://github.com/kubernetes-sigs/cluster-api/blob/v1.8.0/internal/controllers/machine/machine_controller.go#L346).
+
+## Node drain
```

Comment:
I will add a short intro describing that Cluster API is implementing a drain process aligned with the one implemented by kubectl drain + the fact that a controller cannot wait until drain completes, so CAPI version of drain is not blocking and gets continuously retried until success or timeout are reached